Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,490 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Plugin.Calendars.Abstractions;
+using Tizen.Pims.Calendar;
+using Tizen.Pims.Calendar.CalendarViews;
+
+namespace Plugin.Calendars
+{
+	/// <summary>
+	/// Implementation for Calendars
+	/// </summary>
+	public class CalendarsImplementation : ICalendars
+	{
+		/// <summary>
+		/// Gets a list of all calendars on the device.
+		/// </summary>
+		/// <returns>Calendars</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<Calendar>> GetCalendarsAsync()
+		{
+			return Task.Run<IList<Calendar>>(() =>
+			{
+				var calendars = new List<Calendar>();
+				CalendarRecord calRecord = null;
+
+				try
+				{
+					calRecord = new CalendarRecord(Book.Uri);
+					calRecord.Set<string>(Book.Name, "org.tizen.calendar");
+					calendars.Add(
+						new Calendar
+						{
+							Name = calRecord.Get<string>(Book.Name),
+							ExternalID = calRecord.Get<int>(Book.Id).ToString(),
+							CanEditEvents = true,
+							CanEditCalendar = false,
+							AccountName = calRecord.Get<int>(Book.AccountId).ToString(),
+							Color = calRecord.Get<string>(Book.Color),
+						}
+					);
+				}
+				catch (Exception ex)
+				{
+					throw new Exception(ex.Message);
+				}
+				finally
+				{
+					calRecord?.Dispose();
+					calRecord = null;
+				}
+
+				return calendars;
+			});			
+		}
+
+		/// <summary>
+		/// Gets a single calendar by platform-specific ID.
+		/// </summary>
+		/// <param name="externalId">Platform-specific calendar identifier</param>
+		/// <returns>The corresponding calendar, or null if not found</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<Calendar> GetCalendarByIdAsync(string externalId)
+		{
+			CalendarRecord calRecord = null;
+			Calendar calendar = null;
+
+			try
+			{
+				calRecord = new CalendarRecord(Book.Uri);
+				calRecord.Set<string>(Book.Name, "org.tizen.calendar");
+				calendar = new Calendar();
+				calendar.Name = calRecord.Get<string>(Book.Name);
+				calendar.ExternalID = calRecord.Get<int>(Book.Id).ToString();
+				calendar.CanEditEvents = true;
+				calendar.CanEditCalendar = false;
+				calendar.AccountName = calRecord.Get<int>(Book.AccountId).ToString();
+				calendar.Color = calRecord.Get<string>(Book.Color);
+			}
+			catch (Exception ex)
+			{
+				throw new Exception(ex.Message);
```

Comment:
Please remove these catch-all-and-rethrow exception handlers, unless there's a specific exception that you're trying to convert (in which case just catch that exception and wrap it in an appropriate exception).

You don't need `catch` blocks in order to use `finally`.